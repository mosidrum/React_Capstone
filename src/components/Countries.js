import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { filteredCountries } from '../redux/categorySlice';
import { openModal } from '../redux/modalSlice';

const Countries = () => {
  const result = useSelector(filteredCountries);
  const dispatch = useDispatch();
  const saved = JSON.parse(localStorage.getItem('filteredCountries'));

  const handleSubmit = (id) => {
    dispatch(openModal(id));
  };

  if (!result) {
    return (
      <div>
        {saved.map((each) => (
          <ul key={each.alpha3Code}>
            <li>{each.name}</li>
            <button
              type="submit"
              onClick={() => handleSubmit(each.alpha3Code)}
            >
              View more
            </button>
          </ul>
        ))}
      </div>
    );
  }
  return (
    <div>
      {result.map((each) => (
        <ul key={each.alpha3Code}>
          <li>{each.name}</li>
          <button
            type="submit"
            onClick={() => handleSubmit(each.alpha3Code)}
          >
            View more
          </button>
        </ul>
      ))}
    </div>
  );
};

export default Countries;
