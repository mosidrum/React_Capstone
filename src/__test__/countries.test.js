import React from 'react';
import { render, screen } from '@testing-library/react';
import { useSelector } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import Countries from '../components/Countries';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

const mockedCountries = [
  {
    alpha3Code: 'ABC',
    name: 'Country 1',
    flag: 'flag1.png',
    demonym: 'Demonym 1',
    population: 1000,
  },
];

const mockedLocalStorageData = JSON.stringify(mockedCountries);

beforeEach(() => {
  useSelector.mockReturnValue(mockedCountries);
  window.localStorage.__proto__.getItem = jest.fn(() => mockedLocalStorageData);
});

describe('Countries Component', () => {
  test('renders countries correctly', () => {
    render(<Countries />);
    expect(screen.getByText('Country 1')).toBeInTheDocument();
  });

  test('handles country selection', () => {
    render(<Countries />);
    screen.getByText('Country 1').click();
    expect(screen.getByText('Demonym:')).toBeInTheDocument();
  });
});
