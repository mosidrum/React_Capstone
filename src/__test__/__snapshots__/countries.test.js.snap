// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries component should render countries correctly 1`] = `
<DocumentFragment>
  <div
    class="countries-container"
  >
    <div
      class="countries"
    >
      <div
        class="country"
        role="button"
        tabindex="0"
      >
        <img
          alt="Afghanistan"
          src="afghanistan.png"
        />
        <h3>
          Afghanistan
        </h3>
        <div
          class="bottom"
        >
          <li>
            Demonym:
            <span
              class="value"
            >
              Afghan
            </span>
          </li>
          <li>
            Population:
            <span
              class="value"
            >
              10000000
            </span>
          </li>
          <button
            type="submit"
          >
            View more
          </button>
        </div>
      </div>
      <div
        class="country"
        role="button"
        tabindex="0"
      >
        <img
          alt="United States"
          src="usa.png"
        />
        <h3>
          United States
        </h3>
        <div
          class="bottom"
        >
          <li>
            Demonym:
            <span
              class="value"
            >
              American
            </span>
          </li>
          <li>
            Population:
            <span
              class="value"
            >
              300000000
            </span>
          </li>
          <button
            type="submit"
          >
            View more
          </button>
        </div>
      </div>
    </div>
    <div
      class="details"
    />
  </div>
</DocumentFragment>
`;
